src/containers/profile/accountTab/index.jsx

```diff
@@ -1,47 +1,140 @@
-import React, { useState } from 'react';
-import styles from './styles.module.scss';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { isEmpty } from 'lodash';
 
 import { Profile } from '../profile';
 
-export const AccountTab = () => {
-  const [update, setUpdate] = useState(true);
-  const {
-    account,
-    password,
-    buttons_block,
-    payment_btn,
-    payment_btn_block,
-  } = styles;
-
-  const updatePassword = () => {
-    setUpdate(!update);
+import { changePasswordValidate } from '../../../validation/auth';
+import { changePassword } from '../../../services/auth';
+
+import styles from './styles.module.scss';
+
+class AccountTab extends Component {
+  state = {
+    password: '',
+    confirmPassword: '',
+    update: true,
+    errors: {},
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+    const {
+      user: {
+        user: { id },
+      },
+    } = this.props;
+    const { password, confirmPassword } = this.state;
+
+    const errors = changePasswordValidate(password, confirmPassword);
+
+    if (!isEmpty(errors)) {
+      this.setState({ errors });
+    } else {
+      this.setState({ update: !this.state.update });
+      this.setState({ 
+        password: '',
+        confirmPassword: '',
+        errors: {} 
+      });
+      changePassword(id, { password });
+    }
+  };
+
+  onChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  updatePassword = () => {
+    this.setState({ update: !this.state.update });
+  };
+
+  closeUpdate = () => {
+    this.setState({ 
+      password: '',
+      confirmPassword: '',
+      update: !this.state.update,
+      errors: {} 
+    });
   }
 
-  return ( 
-    <>
-      <Profile />
-      <div className={account}>
-        <form>
-          <div className={password}>
-            <input type="text" placeholder="password" disabled={update} />
-          </div>
-          <div className={password}>
-            <input type="text" placeholder="confirm password" disabled={update} />
-          </div>
-          <div className={buttons_block}>
-            {update && <button onClick={updatePassword}>update password</button>}
-            {!update && (
-              <>
-                <button type='submit'  onClick={updatePassword}>save</button>
-                <button type='button' onClick={updatePassword}>close</button>
-              </>
-            )}
+  render() {
+    const { password, confirmPassword, errors, update } = this.state;
+    const {
+      account,
+      pass,
```

Comment:
pls rename this field. I think pass isn't correct class name.

Reply:
done